webapp/src/main/java/sunrise/pokedex/springmvc/manager/PokemonManagerImpl.java

```diff
@@ -0,0 +1,121 @@
+package sunrise.pokedex.springmvc.manager;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import sunrise.pokedex.springmvc.view.PokemonViewImpl;
+import sunrise.pokedex.springmvc.model.Pokemon;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+public class PokemonManagerImpl implements PokemonManager {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private String providerUrl = "http://localhost:8081/pokemon/";
```

Comment:
Don't hardcode this. A good place to store this would be in application.properties file and pull it in.
[stackoverflow.com/questions/30528255/how-to-access-a-value-defined-in-the-application-properties-file-in-spring-boot](https://stackoverflow.com/questions/30528255/how-to-access-a-value-defined-in-the-application-properties-file-in-spring-boot)